src/charm.py

```diff
@@ -281,7 +323,14 @@ def _on_start(self, event) -> None:
                 event.defer()
                 return
 
-        self._connect_mongodb_exporter()
+        try:
+            self._connect_mongodb_exporter()
+        except ChangeError as e:
+            logger.error(
+                "An exception occurred when starting mongodb exporter, error: %s.", str(e)
+            )
+            self.unit.status = BlockedStatus("couldn't start mongodb exporter")
```

Comment:
Should this event be deferred in this case?

Reply:
I don't think so, this  behaviour is same as in VM charm, but just catches different exception